SubEthaEdit-Mac/Source/TabbedDocument.m

```diff
@@ -0,0 +1,145 @@
+//
+//  TabbedDocument.m
+//  SubEthaEdit
+//
+//  Created by Francisco Tolmasky on 8/30/19.
+//  Copyright © 2019 SubEthaEdit Contributors. All rights reserved.
+//
+
+#import "TabbedDocument.h"
+
+
+static __auto_type windowHasAttachedSheet =
+ ^ (NSWindow *window, NSUInteger index, BOOL *stop) {
+        return (BOOL)(window.attachedSheet != nil);
+};
+
+@implementation TabbedDocument
+
+// As we have not implemented document-based alert queueing yet, this simply returns
+// YES if any of it's associated windows has a sheet attached. The eventual theory is
+// that if *any* window has a sheet, they (conceptually) *all should*. In a following
+// commit, this will be expanded to the possibility that no window is *actively*
+// showing a sheet, but may have one queued (in the case where every tab representing
+// the document is hidden).
+- (BOOL)hasAlerts {
+    NSArray *windows = [self.windowControllers valueForKey:@"window"];
+    NSUInteger index = [windows indexOfObjectPassingTest:windowHasAttachedSheet];
```

Comment:
if you really want to do it with indexOfObjectPassingTest, please put the static inside the function. (I would prefer a simpler iteration with early return for this simple kind of test vs. two heavy handed arrays and tests. E.g.
```objective-c
for (windowController in self.windowControllers) {
  if (windowController.window.attachedSheet) {
    return YES;
}
}
return NO;
```

has imho better performance and readability.